src/OrchardCore/OrchardCore.Abstractions/Shell/ShellSettings.cs

```diff
@@ -38,6 +38,12 @@ public ShellSettings(ShellSettings settings)
 
         public string Name { get; set; }
 
+        public string Category
+        {
+            get => _settings[nameof(Category)];
+            set => _settings[nameof(Category)] = value;
+        }
```

Comment:
`ShellSettings` holds 2 `ShellConfiguration` config sources

https://github.com/OrchardCMS/OrchardCore/blob/dbd83c6f226c15db5e6ba82b0e2c1609aa693d77/src/OrchardCore/OrchardCore.Abstractions/Shell/ShellSettings.cs#L20-L24

For both of them, a value can be pre-configured e.g. through the main `appsettings.json`, but when a value is added / updated at runtime, e.g. through a setup or while editing a tenant, a value updated in `_configuration` will be persisted in the tenant specific `appsettings.json` under the tenant folder (can be configured to be in an azure blob or database), a value updated in `_settings` will be persisted in the `tenants.json` (shared by all tenants) directly under the `App_Data` folder (can be configured to be elsewhere, in an azure blob or database).

The `tenants.json` file (related to `_settings`) is intended to hold the minimum set of values (not already pre-configured) required to init a tenant, the other values held by tenant specific sources (related to `_configuration`) being used to lazily build a tenant.

So here for `Category` better to not use a dedicated field using `_settings`, but instead the following indexer that uses `_configuration`, as we already do e.g. for the tenant `Description`.

https://github.com/OrchardCMS/OrchardCore/blob/dbd83c6f226c15db5e6ba82b0e2c1609aa693d77/src/OrchardCore/OrchardCore.Abstractions/Shell/ShellSettings.cs#L69-L74

So nothing to do here, and in the other place just do what we already did for the tenant `Description`.

Reply:
Thanks for the info @jtkech, I will use indexer instead